internal/handler/pathParameter_test.go

```diff
@@ -0,0 +1,28 @@
+package handler
+
+import (
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetPathParameter(t *testing.T) {
+	w := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(w)
+	ctx.AddParam("id", "123")
+
+	id, ok := GetPathParameter(ctx, "id")
+	assert.True(t, ok)
+	assert.Equal(t, uint(123), id)
+}
+
+func TestGetPathParameter_NotFound(t *testing.T) {
+	w := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(w)
+
+	id, ok := GetPathParameter(ctx, "id")
```

Comment:
Why? What does that have to do with this? :smile:

Reply:
because you are setting (sending) the status code to 400 in `GetPathParameter`

`_ = c.AbortWithError(http.StatusBadRequest, fmt.Errorf("error parsing %q: %v", parameter, err))`

which is why clients of `GetPathParameter` only return and don't have to set any error like we do in other places.

Reply:
Yes, of course 👍🏻